lib/serializer-utils.js

```diff
@@ -181,8 +181,10 @@ module.exports = function (collectionName, record, payload, opts) {
       if (opts.includedLinks) {
         included.links = getLinks(dest, opts.includedLinks);
       }
-
-      pushToIncluded(payload, included);
+      // only add to included array if the resource is defined
+      if(id !== 'undefined') {
```

Comment:
I think it would be better if the `getRef` function to get the `id` here returns `undefined` and not `'undefined'`.
Then, the condition will be:

`      if (typeof id !== 'undefined') { pushToIncluded(payload, included); }`